File: CS/DxSample/MainForm.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using DevExpress.Data;
using DevExpress.XtraEditors;
using DxSample.Filtering;

namespace DxSample {
    public partial class MainForm : XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        private void MainForm_Load(object sender, EventArgs e) {
            DataSet ds = new DataSet();
            ds.ReadXml("nwind.xml");
            DataTable dt = ds.Tables["Customers"];
            var customers = from r in dt.Rows.Cast<DataRow>()
                            let required = new string[] {
                                "ANTON", "BERGS", "BLONP", "BOLID", "COMMI",
                                "FOLKO", "GALED", "GODOS", "HILAA"
                            }
                            where required.Contains((string)r["CustomerID"])
                            select new {
                                CompanyName = (string)r["CompanyName"],
                                ContactName = (string)r["ContactName"],
                                ContactTitle = (string)r["ContactTitle"]
                            };
            this.GridControl.DataSource = customers.ToList();
        }

        private void GridView_SubstituteFilter(object sender, SubstituteFilterEventArgs e) {
            e.Filter = GridFilterSubstitutor.Substitute(e.Filter);
        }
    }
}

File: VB/DxSample/MainForm.vb
Imports System
Imports System.Data
Imports System.Linq
Imports DevExpress.Data
Imports DevExpress.XtraEditors
Imports DxSample.Filtering

Namespace DxSample

    Public Partial Class MainForm
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim ds As DataSet = New DataSet()
            ds.ReadXml("nwind.xml")
            Dim dt As DataTable = ds.Tables("Customers")
            Dim customers = From r In dt.Rows.Cast(Of DataRow)() Let required = New String() {"ANTON", "BERGS", "BLONP", "BOLID", "COMMI", "FOLKO", "GALED", "GODOS", "HILAA"} Where required.Contains(CStr(r("CustomerID"))) Select New With {.CompanyName = CStr(r("CompanyName")), .ContactName = CStr(r("ContactName")), .ContactTitle = CStr(r("ContactTitle"))}
            GridControl.DataSource = customers.ToList()
        End Sub

        Private Sub GridView_SubstituteFilter(ByVal sender As Object, ByVal e As SubstituteFilterEventArgs)
            e.Filter = GridFilterSubstitutor.Substitute(e.Filter)
        End Sub
    End Class
End Namespace
